doc/make.jl

```diff
@@ -380,13 +380,20 @@ function Documenter.Writers.HTMLWriter.expand_versions(dir::String, v::Versions)
     vnums = [VersionNumber(x) for x in available_folders]
     master_version = maximum(vnums)
     filter!(x -> x.major == 1 && x.minor == master_version.minor-1, vnums)
-    rc = maximum(vnums)
-    if !isempty(rc.prerelease) && occursin(r"^rc", rc.prerelease[1])
-        src = "v$(rc)"
-        @assert src ∈ available_folders
-        push!(v.versions, src => src, pop!(v.versions))
+    # Include the latest release candidate (rc) or pre-release (alphas and betas)
+    prereleases = filter(x -> !isempty(x.prerelease), vnums)
+    rc_found = false
+    for rc_type in ("rc", "beta", "alpha")
+        for rc in prereleases
+            if occursin(Regex("^$(rc_type)"), rc.prerelease[1])
+                src = "v$(rc)"
+                @assert src ∈ available_folders
+                push!(v.versions, src => src, pop!(v.versions))
+                rc_found = true
+            end
+        end
+        rc_found && break
```

Comment:
@Jay-sanjay Sorry that the PR has been stuck this long. Let's try to get this merged. But I personally can't really reliably reason about the code here.

Here's what I suggest: let's factor this whole bit into a separate function that takes `available_folders` as input, and outputs an array that corresponds to `v.versions`. And then let's add an `@testset` into the `make.jl` here with a bunch of test cases that test that function, to make sure it behaves correctly (e.g. what if there are no pre-releases; there are rc-only; beta and rc; alpha, beta, rc; alpha only; etc). Once it's clear that this loop produces the correct list, then we should be good to go.

To figure out what `available_folders` should look like in a real case, you can look at the directory here -- I believe that is what it `readdir`s: https://github.com/JuliaLang/docs.julialang.org/tree/gh-pages/en